### PR TITLE
Enrich articles with extra content

### PR DESCRIPTION
**Pull request by leogrim**
Gives Extractors access to HttpFetcher so that they can fetch extra content

@gmethvin Feel free to comment on code structure :)
@ymatsuda : 
1. Is there a reason why Scraper is not a Singleton?
2. The idea here is that an extractor can make extra calls to enrich its content before it returns it (e.g. extra article pages, YouTube CC). As discussed, this is opposed to the concept of having articles link to other articles. The main consequence is that this does not allow us to weigh this extra content from peripheral sources any differently that the content scraped from the central page.

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/scraper/extractor/Extractor.scala [hunk 1]
```diff
@@ -1,16 +1,39 @@
...
 import com.keepit.common.net.URI
+import scala.util.{Failure, Success}
+import com.keepit.common.logging.Logging
+import com.google.inject.{Singleton, Inject}
 
-object Extractor {
-  val factories = Seq(
-    YoutubeExtractorProvider,
+@Singleton
+class ExtractorFactories @Inject() (httpFetcher: HttpFetcher) extends Logging {
```
Comment (gmethvin) [inline kifi-backend/modules/shoebox/app/com/keepit/scraper/extractor/Extractor.scala]: Could you just call this `ExtractorFactory` or maybe `ExtractorProvider`? You could also make it a function `String => Extractor`.

**Comment by ymatsuda** [comment]
I think it is ok to make it a singleton.

**Comment by leogrim** [comment]
@ymatsuda 
It works! (modulo a possible need to use a proxy in production)
If multiple tracks are available, here's the priority selection:
1. Track marked as default
2. Non-automatic track in the same language as the automatic track (= language spoken in the video)
3. Automatic track
4. English track

**Comment by eishay** [comment]
wow, cool!
Good job

**Pull request merged**